src/python/grpcio/grpc/_cython/_cygrpc/aio/callback_common.pyx.pxi

```diff
@@ -23,11 +24,19 @@ cdef class CallbackFailureHandler:
         self._core_function_name = core_function_name
         self._error_details = error_details
         self._exception_type = exception_type
+        self._core_error_string = NULL
+        if isinstance(exception_type, enum.IntEnum):
+            self._core_error_string = grpc_call_error_to_string(exception_type)
 
     cdef handle(self, object future):
-        future.set_exception(self._exception_type(
-            'Failed "%s": %s' % (self._core_function_name, self._error_details)
-        ))
+        if self._core_error_string == NULL:
```

Comment:
After some digging, it looks like this `CallbackFailureHandler` actually handles the case that `grpc_event.success` is 0, not the case that we have a bad `grpc_call_error`. We'll want to revert the changes to this class.

Reply:
You're right, I'll revert changes to `CallbackFailureHandler`.